src/ser.rs

```diff
@@ -162,7 +164,14 @@ impl ser::SerializeMap for MapSerializer<'_> {
 
     fn end(self) -> Result {
         debug_assert!(self.next_key.is_none());
-        Ok(self.target.into())
+        if self.as_object {
+            Ok(
+                js_sys::Object::from_entries(&Array::from(&self.target.entries().into()).into())?
```

Comment:
Not sure if this is fine.
- There is no method to set a property on a `js_sys::Object`
- Should the entries be collected in a `Vec<Array>` and then `Object::from_entries` be called at the end?

Reply:
> There is no method to set a property on a js_sys::Object

Yeah js-sys recommends using `Reflect` bindings for that, but it's... not ideal IMO. That's why I'm using custom bindings that generate plain indexing access instead, like you would in JS. See line 10 in this file for bindings themselves and take a look at how structs are already serialized to JS objects - you'd likely want to reuse the same APIs.

> Should the entries be collected in a Vec<Array> and then Object::from_entries be called at the end?

If you go down this route, then note that `Object.fromEntries` accepts arbitrary iterables of `[key, value]` tuples, not just arrays. `Map` conforms to this iterable protocol too and emits same tuples, so you can literally do (Rust equivalent of) `Object.fromEntries(jsMap)`, without conversion to an intermediate array.

However, this approach is not as efficient in terms of both speed and code size, and I'd still prefer to build JS object right away without intermediate Map.